common/lib/xmodule/xmodule/modulestore/split_mongo/caching_descriptor_system.py

```diff
@@ -131,6 +131,13 @@ def _load_item(self, usage_key, course_entry_override=None, **kwargs):
 
         class_ = self.load_block_type(block_data.block_type)
         block = self.xblock_from_json(class_, course_key, block_key, block_data, course_entry_override, **kwargs)
+
+        # TODO Once PLAT-1055 is implemented, we can expose the course version
+        # information within the key identifier of the block.  Until then, set
+        # the course_version as a field on the returned block so higher layers
+        # can use it when needed.
+        block.course_version = version_guid
```

Comment:
Apologies for the clutter on this thread- I believe that test was passing because it does not use the caching layer. Subsection grades' course versions were always null because the course version was not populated in the fallback code that retrieves the block from the modulestore if the block is not in the cache. That's different code from the split.py `_load_items` function (which does populate `course_version`). Why the two paths don't converge on the same function is another question entirely. To me it seems that if the cache does not contain the block, we should call the split modulestore's method to retrieve the item and cache the result, but that's not how it currently works.